shields: nxp: lcd_par_s035: fix gt911 touch initialization

DerekSnell opened the pull request:
Fixes #98876 

Defers init of GT911 touch, MIPI DBI, and display device drivers.  `lcd_par_s035_init()` in shield.c now initializes these to control the GT911 INT_GPIO state during the Reset sequence.

- Fixes issue with this shield caused by shared Reset signal between the GT911 and display driver.
- Removes `reset-gpios` property from `gt911` node.  The GT911 driver does not need to assert the Reset signal because the display driver asserts Reset before the GT911 driver initializes.

Tested this fix on the following:
- LCD_PAR_S035 shield with FRDM_MCXN947
- LCD_PAR_S035 shield with FRDM_RW612

This capture is from the LCD_PAR_S035 with the FRDM_RW612 and this change applied.  The GT911 INT_GPIO is driven low first, then the LCDIC display driver asserts Reset and INT_GPIO remains low through the init process.
<img width="1717" height="355" alt="image" src="https://github.com/user-attachments/assets/31de8746-3f05-4235-b596-923dfc7940cd" />

JarmouniA commented on drivers/input/input_gt911.c:
```diff
@@ -434,6 +434,38 @@ static int gt911_init(const struct device *dev)
 	return 0;
 }
 
+#if CONFIG_INPUT_GT911_PREPARE_INT_GPIO
```
```suggestion
#ifdef CONFIG_INPUT_GT911_PREPARE_INT_GPIO
```

JarmouniA reviewed (dismissed):
Can't you [deffer the initialization](https://docs.zephyrproject.org/latest/kernel/drivers/index.html#deferred-initialization) of display & touch controllers and do the setup in the right order in the board's (or shield's, if that's a thing) `board.c`? 
I really do not like seeing code in the touch controller driver that actually has nothing to do with it, and it's solving a problem created by the shield.

decsny commented:
> Can't you [deffer the initialization](https://docs.zephyrproject.org/latest/kernel/drivers/index.html#deferred-initialization) of display & touch controllers and do the setup in the right order in the board's (or shield's, if that's a thing) `board.c`? I really do not like seeing code in the touch controller driver that actually has nothing to do with it, and it's solving a problem created by the shield.

I don't see how deferred initialization fixes this. Whether you defer or not you still have to have the same patch @DerekSnell submitted here.

Commit:
shields: nxp: lcd_par_s035: fix gt911 touch initialization

Defers init of GT911 touch, MIPI DBI, and display device drivers.
lcd_par_s035_init() in shield.c now initializes these to control the
GT911 INT_GPIO state during the Reset sequence.

Signed-off-by: Derek Snell <derek.snell@nxp.com>
